////
package AlarmMsg;
////
//@服务器类型
enum ServerType {
    ALARM_CLIENT = 37;
    ALARM_SERVER = 38;
}

//@已经发现的质数和处理时间
message PastPrimeNum
{
    required int32 prime_num = 1;
    required int32 process_time = 2;
}

//@信令ID常量定义
enum AlarmId {
    AUTHORIZE_REQ = 1;
    AUTHORIZE_RSP = 2;
    CHECK_PRIME_REQ = 3;
    CHECK_PRIME_RSP = 4;
}

//@公用信令头
message Alarm {
    required string SvcType = 1; //模块名
	required string dwMsgType = 2; //消息类型
	required string alarmType = 3;//服务类型
	optional string userid = 4;	//用户
	optional string channel = 5;//通道
	optional string alarmStatus = 6;//报警状态 1报警 0 非报警
	optional string dwRev = 7;//导致报警操作类型	
	optional string alarmInfo = 8;//报警信息
	optional string alarmhost = 9;//报警ip地址
}

//@报警请求
message AuthorizeReq {
	required Alarm alarmInfo = 1;
	}
	
//@报警响应
message AuthorizeRsp
{
    required Alarm alarmInfo = 1; 
}

//@验证素数请求
message CheckPrimeReq
{
    required Alarm alarmInfo = 1;
}

//@验证素数响应
message CheckPrimeRsp
{
    required Alarm Alarm = 1;
    required int32 number_checked = 2; 
    optional int32 process_time = 33;  
    repeated PastPrimeNum past_prime_num_list = 65;
}


	